src/OpenCensus.Collector.Implementation.Common/PropertyFetcher.cs

```diff
@@ -33,7 +33,14 @@ public object Fetch(object obj)
         {
             if (this.innerFetcher == null)
             {
-                this.innerFetcher = PropertyFetch.FetcherForProperty(obj.GetType().GetTypeInfo().GetDeclaredProperty(this.propertyName));
+                var type = obj.GetType().GetTypeInfo();
+                var property = type.GetDeclaredProperty(this.propertyName);
+                if (property == null)
+                {
+                    property = type.GetProperty(this.propertyName);
```

Comment:
are we doing private reflection here?

Reply:
It's not private reflection. It's getting property of base class